File: main/src/main/java/com/sedmelluq/discord/lavaplayer/natives/samplerate/SampleRateLibrary.java
package com.sedmelluq.discord.lavaplayer.natives.samplerate;

import com.sedmelluq.discord.lavaplayer.natives.ConnectorNativeLibLoader;

class SampleRateLibrary {
  private SampleRateLibrary() {

  }

  static SampleRateLibrary getInstance() {
    ConnectorNativeLibLoader.loadConnectorLibrary();
    return new SampleRateLibrary();
  }

  native long create(int type, int channels);

  native void destroy(long instance);

  native void reset(long instance);

  native int process(long instance, float[] in, int inOffset, int inLength, float[] out, int outOffset, int outLength, boolean endOfInput, double sourceRatio, int[] progress);
}

File: main/src/main/java/com/sedmelluq/discord/lavaplayer/natives/samplerate/SampleRateLibrary.kt
package com.sedmelluq.discord.lavaplayer.natives.samplerate

import com.sedmelluq.discord.lavaplayer.natives.ConnectorNativeLibLoader


internal class SampleRateLibrary private constructor() {
    external fun create(type: Int, channels: Int): Long

    external fun destroy(instance: Long)

    external fun reset(instance: Long)

    external fun process(
        instance: Long,
        `in`: FloatArray?,
        inOffset: Int,
        inLength: Int,
        out: FloatArray?,
        outOffset: Int,
        outLength: Int,
        endOfInput: Boolean,
        sourceRatio: Double,
        progress: IntArray?
    ): Int

    companion object {
        @kotlin.jvm.JvmStatic
        val instance: SampleRateLibrary
            get() {
                ConnectorNativeLibLoader.loadConnectorLibrary()
                return SampleRateLibrary()
            }
    }
}
